pkg/healthcheck/healthcheck.go

```diff
@@ -1147,12 +1147,17 @@ func (hc *HealthChecker) allCategories() []category {
 							return err
 						}
 
+						outdatedPods := []string{}
 						for _, pod := range pods {
 							err = hc.latestVersions.Match(pod.ProxyVersion)
 							if err != nil {
-								return fmt.Errorf("%s: %s", pod.Name, err)
+								outdatedPods = append(outdatedPods, fmt.Sprintf("\t* %s (%s)", pod.Name, pod.ProxyVersion))
 							}
 						}
+						if len(outdatedPods) > 0 {
+							podList := strings.Join(outdatedPods, "\n")
+							return fmt.Errorf("Some data plane pods are not running the current version:\n%s", podList)
+						}
```

Comment:
WDYT about applying the same principle in the check below ("data plane and cli versions match") so it doesn't short-circuit on the first error?
```
linkerd-data-plane
------------------
√ data plane namespace exists
√ data plane proxies are ready
√ data plane proxy metrics are present in Prometheus
‼ data plane is up-to-date
    Some data plane pods are not running the current version:
        * emojivoto/emoji-5fbd6b5f9b-ql8zl (edge-20.3.4)
        * emojivoto/vote-bot-685b789786-mhmqv (edge-20.3.4)
        * emojivoto/voting-54cfd58886-l9vpn (edge-20.3.4)
        * emojivoto/web-776bd8d5cd-fj9kl (edge-20.3.4)
    see https://linkerd.io/checks/#l5d-data-plane-version for hints
‼ data plane and cli versions match
    emojivoto/emoji-5fbd6b5f9b-ql8zl running edge-20.3.4 but cli running git-1b78b85d
    see https://linkerd.io/checks/#l5d-data-plane-cli-version for hints

Status check results are √
```

Reply:
I'm not sure.  This the version match check already feels a bit redundant with the up-to-date check and in many cases the list of affected pods would be the same.  There's room for improvement here but I'm not sure just listing the pods twice is the right thing to do.  Since it's not immediately obvious what to do here, I'd be inclined to leave it alone for now.